### DISCUSSION TITLE
Using DuckDB persistent storage via ODBC

### DISCUSSION THREAD

**Discussion by Locustv2** [duckdb/duckdb#11451] [Q&A]
Hey guys, i followed the [guide](https://duckdb.org/docs/api/odbc/linux) to set up ODBC for duck db.

I tested if the ODBC connection is working using `isql duckdb` and everything works fine. I can create tables etc.

However when i ran the `unixodbc_setup.sh` i specified a `-db` flag to point to a db file.
But when i close isql, after creating some tables, and then open it again, the tables are not available.

When i run `SHOW DATABASES;` on `isql` I still get `memory`.

IT seem that the persistent storage for odbc is not working fine or am i doing something wrong?

Here is a simple Dockerfile to test that:

```
FROM debian:bookworm-slim

RUN mkdir /duckdb

RUN apt-get update

RUN apt-get install wget unzip unixodbc unixodbc-dev odbcinst -y

WORKDIR /tmp

RUN wget https://github.com/duckdb/duckdb/releases/download/v0.10.1/duckdb_odbc-linux-amd64.zip

RUN unzip duckdb_odbc-linux-amd64.zip -d /duckdb

WORKDIR /duckdb

RUN ./unixodbc_setup.sh -s -db /duckdb/db/app.db
```

I also tried to create the `app.db` using duckdb cli and then mount this file in the docker container and it still does not work.

Any help would be much appreciated

**Comment by guenp** [comment]
Hi @Locustv2 , I think this issue is slightly different from the one reported in https://github.com/duckdb/duckdb/issues/11380.
It looks like specifying a DuckDB database via DSN indeed does not work on Linux and Mac OS, and it defaults to `:memory:`. I was able to reproduce this on Mac OS.

If you specify the Database directly via the CLI to `iusql` it should work, e.g.

```
iusql "DRIVER={DuckDB Driver};DATABASE=test.db"
```

**Reply by Locustv2** [reply]
Hey guenp, 

So i found out that it works when i use `iusql` but not when i use `isql`. The difference is that `iusql` has builtin unicode support.

However i also tried with `pyodbc` and it seem to work.

But when i am trying on PHP i couldn't get it to work.
